x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -267,7 +270,8 @@ static Set<String> fieldNames(LogicalPlan parsed) {
         references.removeIf(a -> a instanceof MetadataAttribute || MetadataAttribute.isSupported(a.qualifiedName()));
         Set<String> fieldNames = references.names();
         if (fieldNames.isEmpty()) {
-            return IndexResolver.ALL_FIELDS;
+            // there cannot be an empty list of fields, we'll ask the simplest and lightest one instead: _index
+            return IndexResolver.INDEX_METADATA_FIELD;
```

Comment:
This is the meat of the PR. All the scenarios tested proved that asking here for `_index` instead of `*` is enough. `fieldNames` method identifies well (so far) all the situations where `*` is needed.